### ISSUE TITLE
typing.Final not subscriptable

### ISSUE THREAD

**Issue by bryceschober** [PyCQA/pylint#4052]
### Steps to reproduce
```python
from dataclasses import dataclass
from typing import Final

@dataclass
class constants:
    MY_THING: Final[float] = -12.34
```
### Current behavior
```
Value 'Final' is unsubscriptable
```

### pylint --version output
```
$ pylint --version
pylint 2.5.3
astroid 2.4.1
Python 3.9.1 (default, Dec 11 2020, 09:29:25) [MSC v.1916 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
Hello, I cannot reproduce with pylint's master branch, could you try to install pylint `2.6.0` or the latest master (`pip3 install git@github.com:PyCQA/pylint.git@master#egg=pylint)`, please ? (2.5.3 is old)

**Comment by cdce8p** [comment]
I can reproduce it with Python `3.9.1` and pylint `2.6.0`.

However, I believe this issue was already fixed with: https://github.com/PyCQA/astroid/pull/885, at least it works with the current master branch of pylint and astroid just fine.

Similar to #3950, #3951, #3979

**Comment by Pierre-Sassoulas** [comment]
Thanks for testing with latest astroid @cdce8p. It should be fixed in 2.6.1 then.
